sphinx/ext/intersphinx.py

```diff
@@ -351,6 +355,116 @@ def missing_reference(app: Sphinx, env: BuildEnvironment, node: pending_xref,
     return None
 
 
+class IntersphinxDispatcher(CustomReSTDispatcher):
+    """Custom dispatcher for intersphinx role.
+
+    This enables :intersphinx:***: roles on parsing reST document.
+    """
+
+    def __init__(self, env: BuildEnvironment) -> None:
+        self.env = env
+        super().__init__()
+
+    def role(self, role_name: str, language_module: ModuleType, lineno: int, reporter: Reporter
+             ) -> Tuple[RoleFunction, List[system_message]]:
+        if role_name.split(':')[0] == 'intersphinx':
+            return IntersphinxRole(), []
+        else:
+            return super().role(role_name, language_module, lineno, reporter)
+
+
+class IntersphinxRole(SphinxRole):
+    def run(self) -> Tuple[List[Node], List[system_message]]:
+        role_name = self.get_role_name(self.name)
+        if role_name is None:
+            logger.warning(__('role not found: %s'), self.name,
+                           location=(self.env.docname, self.lineno))
+            return [], []
+
+        result, messages = self.invoke_role(role_name)
+        for node in result:
+            if isinstance(node, pending_xref):
+                node['intersphinx'] = True
+
+        return result, messages
+
+    def get_role_name(self, name: str) -> Optional[Tuple[str, str]]:
+        names = name.split(':')
+        if len(names) == 2:
+            # :intersphinx:role:
```

Comment:
I think `:external:***:` is also good naming for this.

Reply:
I think both are ok, but I lean towards changing to ``external`` as it seems to convey the semantics of what it is in a more direct manner.

Reply:
given that `ref`s are often quite comment, I think there is value in compactness here. For this reason I think `external` is better than `intersphinx`, but I'd also propose that `eref` is used instead of `external` (short for "external ref"). What do folks think about that?

Reply:
See comment below regarding ``eref``.